### ISSUE TITLE
AutoDisconnect crashes game

### ISSUE THREAD

**Issue by MischievousMoo** [ImpactDevelopment/ImpactClient#469]
If I turn on AutoDisconnect and my health goes within the range, or if I die instantly, I will leave the server but also crash the game. It also happens if I join a server where I'm already on the deathscreen or below the specified health.

### What happens?

I leave the game and my client crashes
### When does this happen?

When I join a server or a singleplayer world and I go below the health to disconnect
### Exact steps to reproduce?
1. Turn on AutoDisconnect.
2. Go on a server or singleplayer world.
3. Die or go below the specified health on AutoDisconnect, or alr
4. Leave the server and crash your game

### Game output?

```

[18:42:15] [Server thread/INFO]: <MischievousMoo> Hello
[18:42:15] [main/INFO]: [CHAT] <MischievousMoo> Hello
[18:42:22] [Server thread/INFO]: <MischievousMoo> Time to showcase this autodisconnect bug
[18:42:22] [main/INFO]: [CHAT] <MischievousMoo> Time to showcase this autodisconnect bug
[18:42:36] [Server thread/INFO]: <MischievousMoo> 4.02 health to disconnect
[18:42:36] [main/INFO]: [CHAT] <MischievousMoo> 4.02 health to disconnect
[18:42:41] [Server thread/INFO]: <MischievousMoo> three
[18:42:41] [main/INFO]: [CHAT] <MischievousMoo> three
[18:42:43] [Server thread/INFO]: <MischievousMoo> two
[18:42:43] [main/INFO]: [CHAT] <MischievousMoo> two
[18:42:44] [Server thread/INFO]: <MischievousMoo> tone
[18:42:44] [main/INFO]: [CHAT] <MischievousMoo> tone
[18:42:46] [Server thread/INFO]: <MischievousMoo> I mean one
[18:42:46] [main/INFO]: [CHAT] <MischievousMoo> I mean one
[18:42:49] [Server thread/INFO]: [MischievousMoo: Set own game mode to Survival Mode]
[18:42:49] [main/INFO]: [CHAT] Your game mode has been updated to Survival Mode
[18:42:52] [Server thread/INFO]: MischievousMoo fell from a high place
[18:42:52] [main/INFO]: [CHAT] §8[§9Impact§8]§7 You Died At: §9X: §7834, §9Y: §74, §9Z: §7-1405
[18:42:52] [main/INFO]: [CHAT] MischievousMoo fell from a high place
[18:42:52] [Server thread/INFO]: MischievousMoo lost connection: Disconnected
[18:42:52] [Server thread/INFO]: MischievousMoo left the game
[18:42:52] [Server thread/INFO]: Stopping singleplayer server as player logged out
[18:42:52] [Server thread/INFO]: Stopping server
[18:42:52] [Server thread/INFO]: Saving players
[18:42:52] [Server thread/INFO]: Saving worlds
[18:42:52] [Server thread/INFO]: Saving chunks for level 'AutoDisconnect'/overworld
[18:42:52] [Server thread/INFO]: Saving chunks for level 'AutoDisconnect'/the_nether
[18:42:52] [Server thread/INFO]: Saving chunks for level 'AutoDisconnect'/the_end
[18:42:53] [File IO Thread/INFO]: [STDERR]: java.io.IOException: Stream Closed
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at java.io.RandomAccessFile.seek0(Native Method)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at java.io.RandomAccessFile.seek(RandomAccessFile.java:540)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at ayj.a(SourceFile:323)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at ayj.a(SourceFile:263)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at ayj$a.close(SourceFile:244)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at java.util.zip.DeflaterOutputStream.close(DeflaterOutputStream.java:241)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at java.io.FilterOutputStream.close(FilterOutputStream.java:159)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at java.io.FilterOutputStream.close(FilterOutputStream.java:159)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at aye.b(SourceFile:160)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at aye.a(SourceFile:145)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at bgx.c(SourceFile:37)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at bgx.run(SourceFile:30)
[18:42:53] [File IO Thread/INFO]: [STDERR]: 	at java.lang.Thread.run(Thread.java:745)
[18:42:53] [main/FATAL]: Reported exception thrown!
f: Ticking entity
	at amu.k(SourceFile:1389) ~[amu.class:?]
	at bib.t(SourceFile:1691) ~[bib.class:?]
	at bib.az(SourceFile:1000) ~[bib.class:?]
	at bib.a(SourceFile:419) [bib.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:123) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_25]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_25]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_25]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0_25]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
Caused by: java.lang.NullPointerException
	at me.zero.clarinet.u.lambda$new$0(u.java) ~[u.class:?]
	at me.zero.clarinet.u$$Lambda$202/1519949883.invoke(Unknown Source) ~[?:?]
	at me.zero.alpine.listener.Listener.invoke(Listener.java:77) ~[Listener.class:?]
	at me.zero.alpine.EventManager.lambda$post$6(EventManager.java:98) ~[EventManager.class:?]
	at me.zero.alpine.EventManager$$Lambda$96/1393446924.accept(Unknown Source) ~[?:?]
	at java.util.ArrayList.forEach(ArrayList.java:1249) ~[?:1.8.0_25]
	at me.zero.alpine.EventManager.post(EventManager.java:98) ~[EventManager.class:?]
	at me.zero.client.api.event.CAPIEventManager.post(CAPIEventManager.java:34) ~[CAPIEventManager.class:?]
	at bud.handler$onUpdate$zzd000(SourceFile:1224) ~[bud.class:?]
	at bud.B_(SourceFile) ~[bud.class:?]
	at amu.a(SourceFile:1524) ~[amu.class:?]
	at amu.h(SourceFile:1500) ~[amu.class:?]
	at amu.k(SourceFile:1382) ~[amu.class:?]
	... 10 more

```

### Crash report?

```
---- Minecraft Crash Report ----
// Don't be sad. I'll do better next time, I promise!

Time: 11/12/17 6:42 PM
Description: Ticking entity

java.lang.NullPointerException: Ticking entity
	at me.zero.clarinet.u.lambda$new$0(u.java)
	at me.zero.clarinet.u$$Lambda$202/1519949883.invoke(Unknown Source)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:77)
	at me.zero.alpine.EventManager.lambda$post$6(EventManager.java:98)
	at me.zero.alpine.EventManager$$Lambda$96/1393446924.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at me.zero.alpine.EventManager.post(EventManager.java:98)
	at me.zero.client.api.event.CAPIEventManager.post(CAPIEventManager.java:34)
	at bud.handler$onUpdate$zzd000(SourceFile:1224)
	at bud.B_(SourceFile)
	at amu.a(SourceFile:1524)
	at amu.h(SourceFile:1500)
	at amu.k(SourceFile:1382)
	at bib.t(SourceFile:1691)
	at bib.az(SourceFile:1000)
	at bib.a(SourceFile:419)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at me.zero.clarinet.u.lambda$new$0(u.java)
	at me.zero.clarinet.u$$Lambda$202/1519949883.invoke(Unknown Source)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:77)
	at me.zero.alpine.EventManager.lambda$post$6(EventManager.java:98)
	at me.zero.alpine.EventManager$$Lambda$96/1393446924.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at me.zero.alpine.EventManager.post(EventManager.java:98)
	at me.zero.client.api.event.CAPIEventManager.post(CAPIEventManager.java:34)
	at bud.handler$onUpdate$zzd000(SourceFile:1224)
	at bud.B_(SourceFile)
	at amu.a(SourceFile:1524)
	at amu.h(SourceFile:1500)

-- Entity being ticked --
Details:
	Entity Type: null (me.zero.clarinet.hook.b)
	Entity ID: 0
	Entity Name: MischievousMoo
	Entity's Exact location: 834.25, 4.00, -1405.26
	Entity's Block location: World: (834,4,-1406), Chunk: (at 2,0,2 in 52,-88; contains blocks 832,0,-1408 to 847,255,-1393), Region: (1,-3; contains chunks 32,-96 to 63,-65, blocks 512,0,-1536 to 1023,255,-1025)
	Entity's Momentum: 0.00, 0.00, 0.00
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null
Stacktrace:
	at amu.k(SourceFile:1382)
	at bib.t(SourceFile:1691)
	at bib.az(SourceFile:1000)
	at bib.a(SourceFile:419)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_25, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 423161488 bytes (403 MB) / 603979776 bytes (576 MB) up to 2147483648 bytes (2048 MB)
	JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=16M
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	Launched Version: 1.12.2-Impact_4.1
	LWJGL: 2.9.4
	OpenGL: Intel(R) HD Graphics 520 GL version 4.4.0 - Build 21.20.16.4565, Intel
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: No
	Is Modded: Very likely; Jar signature invalidated
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	CPU: 4x Intel(R) Core(TM) i7-6500U CPU @ 2.50GHz
```

- Impact Version: 4.1 for 1.12.2
- Using Optifine? No
- Enabled mods: Autodisconnect 4.02, Respawn (this bug has still happened without Respawn enabled), Middle click friend, Velocity 0%, Anti overlay, ItemPhysics, Trajectories, CameraClip, ESP Shader, Nametags, ItemSaver, No Rotate, Anti Blind, Tracers, Light
- Operating System: Windows 10

**Comment by mrx12335** [comment]
-> it disconnects you

**Comment by MischievousMoo** [comment]
mrx yes, but it crashes the game in the process

**Comment by ZeroMemes** [comment]
Looks like ItemSaver is the issue.

**Comment by ZeroMemes** [comment]
AutoDisconnect was causing a null player to be passed to anything else that used the player update event, ItemSaver causing the crash was a result of it.

**Comment by ZeroMemes** [comment]
Fixed in 4.2
